kura/org.eclipse.kura.wire.helper.provider/src/main/java/org/eclipse/kura/internal/wire/helper/WireHelperServiceImpl.java

```diff
@@ -144,6 +144,11 @@ public WireSupport newWireSupport(final WireComponent wireComponent,
             return null;
         }
 
+        if (wireComponentRef.getProperty(SERVICE_PID) instanceof java.util.ArrayList) {
+            throw new RuntimeException(String.format(
```

Comment:
Sonar is complaining about this issue https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&branch=PR-4390&id=org.eclipse.kura%3Akura&open=AYaSLhG3gxuJpOzIn5La. changing the exception type to `IllegalArgumentException` should be fine.